envbuilder.go

```diff
@@ -1137,3 +1407,111 @@ func copyFile(src, dst string) error {
 	}
 	return nil
 }
+
```

Comment:
review: below functions are extracted

Reply:
Yeah we probably should; I'd like to simplify the `envbuilder.Run` / `RunCacheProbe` functions as much as possible for future maintenance.